docs/containers/docker-compose.md

```diff
@@ -193,6 +193,76 @@ volumes:
+### Command Customization
+[Command customization](https://code.visualstudio.com/docs/containers/reference#_command-customization) provides various ways to customize the compose up based on your requirement. The following are few sample command customization for compose up command.
+#### Base file and an override file
+Let's assume your workspace has a base compose file (`docker-compose.yml`) and an override file for each environments (`docker-compose.dev.yml`, `docker-compose.test.yml` and `docker-compose.prod.yml`) and you always compose up with base file and an override file. In this case the compose up command can be customized like below. When the compose up command is invoked the ${configuration} will be replaced by the selected file.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "override",
+        "template": "docker-compose -f docker-compose.yml ${configurationFile}  up -d --build",
+    }
+]
+```
+#### Template matching
+Let's assume you have different set of input files for each environments. You could define multiple templates with regular expression match, and the selected file name will be matched against this `match` property and corresponding template will be used.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "dev-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.debug.yml -f docker-compose.dev.yml up -d --build",
+        "match": "dev"
+    },
+    {
+        "label": "test-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.debug.yml -f docker-compose.test.yml up -d --build",
+        "match": "test"
+    },
+    {
+        "label": "prod-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.release.yml -f docker-compose.prod.yml up -d --build",
+        "match": "prod"
+    }
+]
+```
+
+#### Pick a template at runtime
+When invoking compose up command with the following command customization, will give an option to choose a template at runtime and the `${configurationFile}` will be replaced with the selected file.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "dev",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.common.test.yml ${configurationFile} up -d --build"
+    },
+    {
+        "label": "test",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.common.dev.yml ${configurationFile} up -d --build"
+    },
+    {
+        "label": "prod",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.common.prod.yml ${configurationFile} up -d --build"
+    },
+],
+```
+
+### Custom Tasks
+You could define a task like the following to invoke a `docker-compose` command. Please refer [custom task](https://code.visualstudio.com/docs/editor/tasks#_custom-tasks) for more detail on this.
```

Comment:
I would start this out with... “Rather than use command customization you can also define a task…”